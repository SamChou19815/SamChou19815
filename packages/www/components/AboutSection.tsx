import React, { ReactElement } from 'react';

import styles from './AboutSection.module.css';
import ButtonLink from './Common/ButtonLink';
import ConsoleSection from './Common/ConsoleSection';
import { ExpandMoreIcon } from './Common/Icons';
import InformationCard from './InformationCard';

const AboutSection = (): ReactElement => (
  <ConsoleSection
    id="about"
    title="dev-sam about"
    className={styles.AboutSection}
    titleClassName={styles.Title}
  >
    <InformationCard className={styles.InfoCard} />
    <ButtonLink className={styles.ReadMore} href="#projects">
      READ MORE
      <ExpandMoreIcon />
    </ButtonLink>
  </ConsoleSection>
);

export default AboutSection;
