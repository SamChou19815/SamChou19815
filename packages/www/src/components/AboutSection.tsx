import React, { ReactElement } from 'react';

import ButtonLink from './Common/ButtonLink';
import ConsoleSection from './Common/ConsoleSection';
import WwwSvgIcon from './Common/Icons';
import InformationCard from './InformationCard';

const AboutSection = (): ReactElement => (
  <ConsoleSection id="about" title="dev-sam about" className="about-section" titleClassName="title">
    <InformationCard className="info-card" />
    <ButtonLink className="read-more" href="#projects">
      READ MORE
      <WwwSvgIcon iconName="expand-more" />
    </ButtonLink>
  </ConsoleSection>
);

export default AboutSection;
