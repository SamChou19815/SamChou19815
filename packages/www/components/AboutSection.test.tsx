import React from 'react';

import renderer from 'react-test-renderer';

import AboutSection from './AboutSection';

it('AboutSection matches snapshot.', () => {
  const tree = renderer.create(<AboutSection />).toJSON();
  expect(tree).toMatchSnapshot();
});
