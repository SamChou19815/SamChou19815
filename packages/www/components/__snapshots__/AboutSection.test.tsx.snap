// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutSection matches snapshot. 1`] = `
<section
  className="Section AboutSection"
  id="about"
>
  <h3
    className="Title Title"
  >
    <code>
      $Â 
      dev-sam about
    </code>
  </h3>
  <div
    className="card InfoCard"
  >
    <div
      className="card__image"
    >
      <img
        alt="Facebook @ 1 Hacker Way"
        loading="lazy"
        src="/timeline/fb-hacker-way.jpg"
        title="Facebook @ 1 Hacker Way"
      />
    </div>
    <div
      className="card__header"
    >
      <h3>
        Sam Zhou
      </h3>
    </div>
    <div
      className="card__body IconLines"
    >
      <div
        className="IconLine"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5 3h14a2 2 0 0 1 2 2v14a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V5a2 2 0 0 1 2-2m13 2h-2.5A3.5 3.5 0 0 0 12 8.5V11h-2v3h2v7h3v-7h3v-3h-3V9a1 1 0 0 1 1-1h2V5z"
          />
        </svg>
        <span
          className="IconLineText"
        >
          Facebook SWE Intern
        </span>
      </div>
      <div
        className="IconLine"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M20 6h-4V4c0-1.11-.89-2-2-2h-4c-1.11 0-2 .89-2 2v2H4c-1.11 0-1.99.89-1.99 2L2 19c0 1.11.89 2 2 2h16c1.11 0 2-.89 2-2V8c0-1.11-.89-2-2-2zm-6 0h-4V4h4v2z"
          />
        </svg>
        <span
          className="IconLineText"
        >
          Cornell DTI Dev Lead
        </span>
      </div>
      <div
        className="IconLine"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 .3a12 12 0 0 0-3.8 23.4c.6.1.8-.3.8-.6v-2c-3.3.7-4-1.6-4-1.6-.6-1.4-1.4-1.8-1.4-1.8-1-.7.1-.7.1-.7 1.2 0 1.9 1.2 1.9 1.2 1 1.8 2.8 1.3 3.5 1 0-.8.4-1.3.7-1.6-2.7-.3-5.5-1.3-5.5-6 0-1.2.5-2.3 1.3-3.1-.2-.4-.6-1.6 0-3.2 0 0 1-.3 3.4 1.2a11.5 11.5 0 0 1 6 0c2.3-1.5 3.3-1.2 3.3-1.2.6 1.6.2 2.8 0 3.2.9.8 1.3 1.9 1.3 3.2 0 4.6-2.8 5.6-5.5 5.9.5.4.9 1 .9 2.2v3.3c0 .3.1.7.8.6A12 12 0 0 0 12 .3"
          />
        </svg>
        <span
          className="IconLineText"
        >
          Open source contributor
        </span>
      </div>
      <div
        className="IconLine"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M5 13.18v4L12 21l7-3.82v-4L12 17l-7-3.82zM12 3L1 9l11 6 9-4.91V17h2V9L12 3z"
          />
        </svg>
        <span
          className="IconLineText"
        >
          Cornell University
        </span>
      </div>
      <div
        className="IconLine"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M24 14V6H4v36h40V14H24zM12 38H8v-4h4v4zm0-8H8v-4h4v4zm0-8H8v-4h4v4zm0-8H8v-4h4v4zm8 24h-4v-4h4v4zm0-8h-4v-4h4v4zm0-8h-4v-4h4v4zm0-8h-4v-4h4v4zm20 24H24v-4h4v-4h-4v-4h4v-4h-4v-4h16v20zm-4-16h-4v4h4v-4zm0 8h-4v4h4v-4z"
            transform="scale(0.5, 0.5)"
          />
        </svg>
        <span
          className="IconLineText"
        >
          Computer Science
        </span>
      </div>
      <div
        className="IconLine"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M9.4 16.6L4.8 12l4.6-4.6L8 6l-6 6 6 6 1.4-1.4zm5.2 0l4.6-4.6-4.6-4.6L16 6l6 6-6 6-1.4-1.4z"
          />
        </svg>
        <span
          className="IconLineText"
        >
          Coding since 13
        </span>
      </div>
    </div>
    <div
      className="card__footer Links"
    >
      <div
        className="button-group button-group--block"
      >
        <a
          className="button button--secondary Link"
          href="https://blog.developersam.com"
        >
          Blog
        </a>
        <a
          className="button button--secondary Link"
          href="https://github.com/SamChou19815"
        >
          GitHub
        </a>
        <a
          className="button button--secondary Link"
          href="/resume.pdf"
        >
          Resume
        </a>
      </div>
    </div>
  </div>
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-text ReadMore MuiButton-textPrimary"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover ReadMoreLink MuiTypography-colorPrimary"
        href="#projects"
        onBlur={[Function]}
        onFocus={[Function]}
      >
        Read more
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </a>
    </span>
  </button>
</section>
`;
